consensus/parlia/parlia.go

```diff
@@ -750,8 +764,13 @@ func (p *Parlia) snapshot(chain consensus.ChainHeaderReader, number uint64, hash
 		// If we're at the genesis, snapshot the initial state. Alternatively if we have
 		// piled up more headers than allowed to be reorged (chain reinit from a freezer),
 		// consider the checkpoint trusted and snapshot it.
-		// An offset `p.config.Epoch - 1` can ensure getting the right validators.
-		if number == 0 || ((number+1)%p.config.Epoch == 0 && (len(headers) > int(params.FullImmutabilityThreshold))) {
+		// An offset `epochLength - 1` can ensure getting the right validators.
+
+		// Unable to retrieve the exact EpochLength here.
+		// Using lorentzEpochLength instead, assuming `lorentzEpochLength % defaultEpochLength == 0`.
+		// TODO(Nathan)(BEP-524 Phase Two): sync latest EpochLength defined
+		epochLength := uint64(lorentzEpochLength)
```

Comment:
should apply hardfork logic here? maybe use `epochLengthBeforeLorentz`.

By the way, I dont think hardcode `lorentzEpochLength` and `lorentzBlockInterval` is a good way to implement this, please try to find a more general path to it.

Reply:
no need to hardfork here, because only get a header with `validators` is need,
```
	defaultEpochLength   = uint16(200)  // Default number of blocks of checkpoint to update validatorSet from contract
	lorentzEpochLength   = uint16(500)  // Epoch length starting from the Lorentz hard fork
	maxwellEpochLength   = uint16(1000) // Epoch length starting from the Maxwell hard fork
```
so use maxwellEpochLength is always right, because 1000%200==0 and 1000%500==0

Reply:
There are two general approaches to implementation, as far as I know:

- Similar to turnLength, but as discussed, we don’t want to make them governable since they are fundamental.
- Similar to BlobScheduleConfig, but we are unlikely to change the epoch or block interval frequently.

Therefore, I am hardcoding them here.

```
// BlobScheduleConfig determines target and max number of blobs allow per fork.
type BlobScheduleConfig struct {
	Cancun  *BlobConfig `json:"cancun,omitempty"`
	Prague  *BlobConfig `json:"prague,omitempty"`
	Lorentz *BlobConfig `json:"lorentz,omitempty"`
	Verkle  *BlobConfig `json:"verkle,omitempty"`
}
```

Reply:
> should apply hardfork logic here?

need more logic here, updated